app/controllers/util/CustomMetadataUtils.scala

```diff
@@ -62,6 +64,16 @@ class CustomMetadataUtils(allCustomMetadataProperties: List[CustomMetadata]) {
         TextField(property.name, fieldLabel, fieldDescription, property.multiValue, InputNameAndValue("years", property.defaultValue.getOrElse("")), "numeric", isRequired)
       case Text =>
         property.propertyType match {
+          case Defined if property.multiValue =>
```

Comment:
This function is pretty long, and there is the nested `match` statement for the `Text` case. I'd be tempted to abstract out the logic for the `Text` case to a separate function.

Reply:
I have extracted the logic of the `Text` case to a separate function.